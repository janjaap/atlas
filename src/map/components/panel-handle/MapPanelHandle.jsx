import React from 'react';
import PropTypes from 'prop-types';

import CollapseIcon from '../../../../public/images/icon-arrow-down.svg';
import ExpandIcon from '../../../../public/images/icon-arrow-up.svg';

const MapPanelHandle = (props) => (
  <div className={`
    map-panel-handle
    map-panel-handle--${props.isMapPanelHandleVisible ? 'visible' : 'hidden'}
  `}
  >
    <button
      className="map-panel-handle__toggle"
      onClick={props.onMapPanelHandleToggle}
    >
      {props.isMapPanelHandleVisible
        ? <span><CollapseIcon /><span className="u-sr-only">Kaartlagen verbergen</span></span>
        : <span><ExpandIcon /><span className="u-sr-only">Kaartlagen tonen</span></span>
      }
    </button>
    {props.isMapPanelHandleVisible && props.children}
  </div>
);

MapPanelHandle.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  isMapPanelHandleVisible: PropTypes.bool.isRequired,
  onMapPanelHandleToggle: PropTypes.func.isRequired
};

export default MapPanelHandle;
