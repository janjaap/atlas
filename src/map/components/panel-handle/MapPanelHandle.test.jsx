import React from 'react';
import { shallow } from 'enzyme';

import MapPanelHandle from './MapPanelHandle';

describe('MapPanelHandle', () => {
  it('should render children by default', () => {
    const mapPanelHandle = shallow(
      <MapPanelHandle
        isMapPanelHandleVisible
        onMapPanelHandleToggle={() => {}}
      >
        <div className="children-test" />
      </MapPanelHandle>
    );

    expect(mapPanelHandle.find('.children-test').length).toBe(1);
    expect(mapPanelHandle).toMatchSnapshot();
  });

  it('should hide children on click', () => {
    const onButtonClickSpy = jest.fn();
    const mapPanelHandle = shallow(
      <MapPanelHandle
        isMapPanelHandleVisible={false}
        onMapPanelHandleToggle={onButtonClickSpy}
      >
        <div className="children-test" />
      </MapPanelHandle>
    );

    mapPanelHandle.find('.map-panel-handle__toggle').simulate('click');
    expect(onButtonClickSpy).toHaveBeenCalledTimes(1);
    expect(mapPanelHandle.find('.children-test').length).toBe(0);
  });
});
