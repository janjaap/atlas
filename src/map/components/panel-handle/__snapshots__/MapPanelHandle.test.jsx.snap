// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapPanelHandle should render children by default 1`] = `
<div
  className="
    map-panel-handle
    map-panel-handle--visible
  "
>
  <button
    className="map-panel-handle__toggle"
    onClick={[Function]}
  >
    <span>
      <Component />
      <span
        className="u-sr-only"
      >
        Kaartlagen verbergen
      </span>
    </span>
  </button>
  <div
    className="children-test"
  />
</div>
`;
